### PR TITLE
feat(radio): add ionic theme styles

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

The `ionic` theme has not been implemented to `ion-radio`.

## What is the new behavior?

- Added `ionic` theme styles.
- Added `ionic` to the tests.
- Updated file structure.
- Added a focus ring mixin.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Preview (basic)](https://ionic-framework-git-rou-11015-ionic1.vercel.app/src/components/radio/test/basic?ionic:theme=ionic)
[Preview (color)](https://ionic-framework-git-rou-11015-ionic1.vercel.app/src/components/radio/test/color?ionic:theme=ionic)
[Preview (item)](https://ionic-framework-git-rou-11015-ionic1.vercel.app/src/components/radio/test/item?ionic:theme=ionic)
[Preview (label placement)](https://ionic-framework-git-rou-11015-ionic1.vercel.app/src/components/radio/test/label-placement?ionic:theme=ionic)
[Preview (states)](https://ionic-framework-git-rou-11015-ionic1.vercel.app/src/components/radio/test/states?ionic:theme=ionic)

### CODE REVIEW

File: core/src/components/radio/radio.ionic.scss [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+// Ionic Radio Inner Circle: Unchecked
+// -----------------------------------------
+
+.radio-inner {
+  @include globals.border-radius(var(--inner-border-radius));
+
+  width: calc(32% + var(--border-width));
+  height: calc(32% + var(--border-width));
```
Comment (brandyscarney) [inline core/src/components/radio/radio.ionic.scss:75]: Where does 32% come from?

File: core/src/components/radio/radio.ionic.scss [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ * when the label sits at the end.
+ */
+:host(.radio-label-placement-end) .label-text-wrapper {
+  @include globals.margin(null, 0, null, globals.$ionic-space-400);
+}
+
+// Radio Label Placement - Fixed
+// ----------------------------------------------------------------
+
+:host(.radio-label-placement-fixed) .label-text-wrapper {
```
Comment (brandyscarney) [inline core/src/components/radio/radio.ionic.scss:116]: Should this be combined with `:host(.radio-label-placement-start) .label-text-wrapper {`?

File: core/src/themes/ionic/ionic.mixins.scss [hunk 3]
```diff
@@ -41,6 +41,22 @@
...
 }
 
+// Focused mixin to be used on regular elements for the Ionic Theme focused state
+//
+// ex: :host(.ion-focused) .toggle-icon {
+//       @include globals.focused-state();
+//      }
+//
+// --------------------------------------------------
+@mixin focused-state(
```
Comment (brandyscarney) [inline core/src/themes/ionic/ionic.mixins.scss:51]: Would it be less confusing to keep the arguments in the same order as `outline` and also make offset optional / a boolean?

**Review by brandyscarney — COMMENTED** [review]
This is looking good, I mostly have questions on some of the implementation details. 🙂

File: core/src/components/radio/radio.ionic.scss [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  border-color: globals.current-color(base);
+
+  background-color: globals.current-color(base);
+}
+
+// Radio Label
+// ----------------------------------------------------------------
+
+:host(.in-item) .label-text-wrapper {
+  @include globals.margin(globals.$ionic-space-250, null, globals.$ionic-space-250, null);
```
[the remark this answers is not in the document]
Comment (thetaPC) [reply core/src/components/radio/radio.ionic.scss]: Did a double check and there's no point in it. Removed: https://github.com/ionic-team/ionic-framework/pull/29805/commits/d489d6c071aa5732b114ca066891102e26b8de84

File: core/src/components/radio/radio.ionic.scss [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+:host(.in-item.radio-label-placement-stacked) .label-text-wrapper {
+  @include globals.margin(globals.$ionic-space-250, null, globals.$ionic-space-400, null);
```
[the remark this answers is not in the document]
Comment (thetaPC) [reply core/src/components/radio/radio.ionic.scss]: Did a double check and there's no point in it. Removed: https://github.com/ionic-team/ionic-framework/pull/29805/commits/d489d6c071aa5732b114ca066891102e26b8de84

File: core/src/components/radio/radio.ionic.scss (hunk 1 above)
Comment (thetaPC) [reply core/src/components/radio/radio.ionic.scss:68]: The designs show the inner circle to be 8px. It didn't feel correct to hard code it in and thought the best approach was to follow `md`. By using 32%, it calculates the size to be a very close number to 8px. It's like 8.0..px.

File: core/src/components/radio/radio.ionic.scss (hunk 2 above)
Comment (thetaPC) [reply core/src/components/radio/radio.ionic.scss]: https://github.com/ionic-team/ionic-framework/pull/29805/commits/59ea4906cac4aefa26d87a8f5e3a6a5d830ee160

File: core/src/themes/ionic/ionic.mixins.scss (hunk 3 above)
Comment (thetaPC) [reply core/src/themes/ionic/ionic.mixins.scss:51]: https://github.com/ionic-team/ionic-framework/pull/29805/commits/ebbb8cb29f754baa4902bf2a70185b67a4ede06a

File: core/src/themes/ionic/ionic.mixins.scss [hunk 6]
```diff
@@ -41,6 +41,40 @@
...
+
+  @if $color == null {
+    $color: tokens.$ionic-state-focus-1;
+  }
+
+  @if $addOffset {
+    outline-offset: $width;
+  }
+
+  outline: $width $style $color;
```
Comment (brandyscarney) [inline core/src/themes/ionic/ionic.mixins.scss:75]:
```suggestion
  // Use default values if null is passed
  $width: if($width != null, $width, tokens.$ionic-border-size-050);
  $style: if($style != null, $style, tokens.$ionic-border-style-solid);
  $color: if($color != null, $color, tokens.$ionic-state-focus-1);

  outline: $width $style $color;

  @if $addOffset {
    outline-offset: $width;
  }
```

Simplified a bit: [Playground](https://sass-lang.com/playground/#eJx9UctOwzAQvPsrFqmHVor7QMAhFVL/gBMfkMbrspJjV7ENBdR/x9mYJrSlJ6+9M7Mz69VqQs5SLbeuVdhKT18ol4/LEu73h7U4a4ZPg9I7Q6oEPk4IH6qAUrs6erkqocXUEpuGDmSBX1H1mKkAmHyQCm8lXB9ddAgedYEY5jOodsa1J9AfC9yvlHrR2mMoIbQRxQy+0/NiAa8eQaGuognwXpmIHkiDjcYAedhXPtkd+SQ97Uu4e2ZUkVvFPxFm61GGjs3liM33C/YQrxfI+ToBLkcCfC+uRk/cRHYxGLLdCnvn2UJPZMQmRR5WxKs50aTLe+vZnZujOAoxT/Nk/k+Y5xV65m7I1iYqPPvu5OaC15D3ZHeyIaWSpxvsh/2hyJG36ZuuienK+F+/t6SeBqldi2gLYGYn+QPFtfz9)

**Review by brandyscarney — APPROVED** [review]
Looks good, just a suggested change to simplify the mixin a bit. 🙂

**Review by BenOsodrac — APPROVED** [review]

**Pull request merged**